### ISSUE TITLE
Manifest v3

### ISSUE THREAD

**Issue by Sytten** [Noovolari/leapp-browser-extension#1]
Just opening this since I am a bit worried that the extension basically will stop working in the next year as manifest v2 is set to be removed in 2023. Do you guys need APIs not available in v3?

Reading a bit the code, the main issue is the usage of `blocking` to intercept the traffic and inject the right cookie. I think it should be possible to use `updateDynamicRules` with the `tabIds` filter to replace the header. Much more annoying than the current approach, but basically on WS message you have to modify the static rules instead of just storing it for the tab.

**Comment by pethron** [comment]
Thanks for pointing this out, @Sytten. The extension, as it is now, will work until Chrome stops supporting manifest v2 (June 2023); we are still publishing the extensions to give something usable to Leapp's users while we search for an alternative solution and to show our commitment to this specific feature.

We are closely monitoring the development of other extensions that relies on the blocking methods to intercept traffic to update our logic for the manifest v3 and also running spikes and research activities to actively try to find a solution for the extension to work with the v3 manifest.

Since this is one of the core components of Leapp (as we find it speeds up devs considerably), this is one of our top priorities. 

I take the chance for a _call to action_ to anyone with expertise in this field to suggest solutions or point us in the right direction. Any help on this issue would be very welcome.

**Comment by vazome** [comment]
Just leaving an update on Manifest V2 deprecation for the future visitors and anyone interested:

https://developer.chrome.com/blog/resuming-the-transition-to-mv3
> We will begin disabling Manifest V2 extensions in [pre-stable versions of Chrome](https://developer.chrome.com/docs/web-platform/chrome-release-channels) (Dev, Canary, and Beta) as early as June 2024, in Chrome 127 and later. Users impacted by the rollout will see Manifest V2 extensions automatically disabled in their browser and will no longer be able to install Manifest V2 extensions from the Chrome Web Store. Also in June 2024, Manifest V2 extensions will lose their Featured badge in the Chrome Web Store if they currently have one.

TL;DR June 2024
